examples/md-flexible/src/ParallelVtkWriter.h

```diff
@@ -76,17 +80,46 @@ class ParallelVtkWriter {
    */
   int _maximumNumberOfDigitsInIteration;
 
+  /**
+   * Writes the current state of particles into vtk files.
+   * @param currentIteration: The simulations current iteration.
+   * @param autoPasContainer The AutoPas container whose owned particles will be logged.
+   */
+  void recordParticleStates(const int &currentIteration, const autopas::AutoPas<ParticleType> &autoPasContainer);
+
+  /**
+   * Writes the current domain subdivision into vtk files.
+   * @param currentIteration: The simulations current iteration.
+   * @param decomposition: The simulations domain decomposition.
+   */
+  void recordDomainSubdivision(const int &currentIteration, const autopas::Configuration &autoPasConfiguration,
+                               const RegularGridDecomposition &decomposition);
+
+  /**
+   * Calculates the whole extent of the decompositions local domain.
+   * The whole extent defines the space this local domain is occupying in the global domain.
+   * The layout of the returned array is [ xmin, xmax, ymin, ymax, zmin, zmax ].
+   * @param decomposition: The simulations domain decomposition.
+   * @return the whole extent of the local domain.
+   */
+  std::array<int, 6> calculateWholeExtent(const RegularGridDecomposition &domainDecomposition);
```

Comment:
what exactly are xmin etc? IDs?

Ok just looked up `MPI_Cart_get()` what you call ID is actually a coordinate in the decomposition grid. I think it would be more intuitive if you would rename this.